R/analyse.R

```diff
@@ -2,18 +2,113 @@
 
 
 
-#' Title
+#' Analyse Multiple Imputed Datasets
 #'
 #' @description
-#' TODO
+#' This function takes multiple imputed datasets (as generated by
+#' the [impute()] function) and runs an analysis function on
+#' each of them.
 #'
-#' @param imputations TODO
-#' @param fun TODO
-#' @param delta TODO
-#' @param ... TODO
+#' @details
+#' This function works by performing the following steps
 #'
+#' 1. Extract a dataset from the `imputations` object
+#' 2. Apply any delta adjustments as specified by the `delta` argument
+#' 3. Run the analysis function `fun` on the dataset
+#' 4. Repeat steps 1-3 across all of the datasets inside the `imputations`
+#' object
+#' 5. Collect and return all of the analysis results.
+#'
+#' The analysis function `fun` must take a data.frame as its first
+#' argument. All other options to [analyse()] are passed onto `fun`
+#' via `...`.
+#' `fun` must return a named list with each element being itself a
+#' list containing a single
+#' numeric element called `est` (or additionally `se` and `df` if using
+#' you had originally specified [method_bayes()] or [method_approxbayes()])
+#' i.e.
+#' ```
+#' myfun <- function(dat, ...) {
+#'     mod_1 <- lm(data = dat, outcome ~ group)
+#'     mod_2 <- lm(data = dat, outcome ~ group + covar)
+#'     x <- list(
+#'         trt_1 = list(
+#'             est = coef(mod_1)[[group]],
+#'             se = sqrt(vcov(mod_1)[group, group]),
+#'             df = df.residual(mod_1)
+#'         ),
+#'         trt_2 = list(
+#'             est = coef(mod_2)[[group]],
+#'             se = sqrt(vcov(mod_2)[group, group]),
+#'             df = df.residual(mod_2)
+#'         )
+#'      )
+#'      return(x)
+#'  }
+#' ```
+#'
+#' Please note that the `vars$subjid` column (as defined in the original call to
+#' [draws()]) will be scrambled in the data.frames that are provided to `fun`.
+#' This is to say they will not contain the original subject values and as such
+#' any hard coding of subject ids is strictly to be avoided.
+#'
+#' By default `fun` is the [ancova()] function.
+#' Please note that this function
+#' requires that a vars object, as created by [set_vars()], is provided via
+#' the `vars` arguement. i.e. `analyse(imputeObj, vars = set_vars(...))` Please
```

Comment:
"argument". I.e. `analyse(imputeObj, vars = set_vars(...))`. Please.....

Reply:
Sorry I am not quite sure I understand what you are referring to here

Reply:
arguement -> argument 
i.e -> I.e
analyse(imputeObj, vars = set_vars(...)) Please -> analyse(imputeObj, vars = set_vars(...)). Please

Reply:
Arrr gotcha.  And done :)